samples/android/app/src/androidTest/java/org/mozilla/samples/gleancore/pings/DeletionRequestPingTest.kt

```diff
@@ -43,13 +43,17 @@ class DeletionRequestPingTest {
         val device = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
         device.waitForIdle()
 
-        // Disable upload by toggline the switch
+        // Wait for any ping to make sure there are no pending requests before going forward
+        waitForPingContent("", server)
```

Comment:
Do we expect to receive any ping here? If not will this cause a 20s delay because it waits for the timeout?

Reply:
Also why can we use `""`? Doesn't that make the waitForPingcontent` spin 3 times, each with a 20s timeout?

Reply:
> Do we expect to receive any ping here? If not will this cause a 20s delay because it waits for the timeout?

The comment we had previously in this stated that there could be events or metrics ping.

> Also why can we use ""? Doesn't that make the waitForPingcontent` spin 3 times, each with a 20s timeout?

I can remove the timeout time from this `takeRequest` function call. Also, I did this by mistake initially and then undid it, but I think it makes sense to just break out of the while loop in case `takeRequest` returns `null`.

Reply:
This timeout parameter is optional actually. I can just remove it.

Reply:
Ah, so if there is something, we get it, drop it, and then wait again for a potential next ping.
So if there are multiple pings getting sent we should be quick through the loop as we don't need to wait, otherwise we most likely wait once to not receive anything, then should drop out after the first timeout.

Reply:
Yes, still 20s is a long wait time. Even if we only wait once.

Reply:
If 20s is a concern, we should test how low we can get. CI can be very slow.

Reply:
Well, I changed that `continue` to a `break`. 

MockWebServer is just a LinkedBlokingQueue of requests. We wait in case there is some background task running that still hasn't finished and might send a ping request, correct?

There is a way to run WorkManager tasks synchronously for integrations tests (see: https://developer.android.com/topic/libraries/architecture/workmanager/how-to/integration-testing), which could be a solution. I am not sure if it is worth it though.

Reply:
> There is a way to run WorkManager tasks synchronously for integrations tests (see: https://developer.android.com/topic/libraries/architecture/workmanager/how-to/integration-testing), which could be a solution. I am not sure if it is worth it though.

That's definitely helpful, but last time we tried it was riddled of bugs and basically not very useful at all :) Things have probably changed!